CMakeLists.txt

```diff
@@ -1,29 +1,136 @@
+	${SENTRY_NATIVE_SOURCE_ROOT}/transports/sentry_function_transport.c
+)
+
+file(GLOB SENTRY_NATIVE_LIBCURL_TRANSPORT_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/transports/sentry_libcurl_transport.c
+)
+
+file(GLOB SENTRY_NATIVE_INPROC_BACKEND_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/backends/sentry_inproc_*.c
+)
+
+file(GLOB SENTRY_NATIVE_UNIX_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/unix/*.c
+)
+
+file(GLOB SENTRY_NATIVE_LINUX_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/linux/*.c
+)
+
+file(GLOB SENTRY_NATIVE_DARWIN_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/darwin/*.c
+)
+
+file(GLOB SENTRY_NATIVE_ANDROID_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/linux/sentry_procmaps_modulefinder.c
+	${SENTRY_NATIVE_SOURCE_ROOT}/unix/sentry_unix_unwinder_libunwindstack.cpp
+)
+
+file(GLOB SENTRY_NATIVE_WINDOWS_SOURCES
+	${SENTRY_NATIVE_SOURCE_ROOT}/windows/*.c
 )
 
-add_definitions(-DSENTRY_WITH_INPROC_BACKEND)
+file(GLOB_RECURSE SENTRY_NATIVE_TEST_SOURCES
+	${CMAKE_CURRENT_SOURCE_DIR}/tests/*.c
+)
+
+set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_SOURCES})
+if(WIN32)
+	set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_ALL_SOURCES}
+		${SENTRY_NATIVE_WINDOWS_SOURCES})
+else()
+	set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_ALL_SOURCES}
+		${SENTRY_NATIVE_UNIX_SOURCES} ${SENTRY_NATIVE_INPROC_BACKEND_SOURCES})
+endif()
+
+if(CURL_FOUND)
+	set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_ALL_SOURCES}
+		${SENTRY_NATIVE_LIBCURL_TRANSPORT_SOURCES})
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_ALL_SOURCES}
+		${SENTRY_NATIVE_DARWIN_SOURCES})
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+	set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_ALL_SOURCES}
+		${SENTRY_NATIVE_LINUX_SOURCES})
+
+	set(LINK_LIBRARIES ${LINK_LIBRARIES} "pthread" "dl")
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Android")
+	set(SENTRY_NATIVE_ALL_SOURCES ${SENTRY_NATIVE_ALL_SOURCES}
+			${SENTRY_NATIVE_ANDROID_SOURCES})
+
+	include_directories(${CMAKE_CURRENT_SOURCE_DIR}/external/libunwindstack-ndk/include)
+	add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/libunwindstack-ndk/cmake)
+	set(LINK_LIBRARIES ${LINK_LIBRARIES} "unwindstack")
+
+endif()
+
+if(WIN32)
+	set(LINK_LIBRARIES ${LINK_LIBRARIES} "dbghelp.lib")
+endif()
+
 add_library("sentry" SHARED
-    ${SENTRY_NATIVE_SOURCES}
+	${SENTRY_NATIVE_ALL_SOURCES}
 )
 target_link_libraries("sentry" ${LINK_LIBRARIES})
+
+# compile tests separately and pass an extra preprocessor define so we can
+# switch some internal modes for the unittests.
+add_executable("sentry_tests"
+	${SENTRY_NATIVE_ALL_SOURCES}
+	${SENTRY_NATIVE_TEST_SOURCES}
+)
+target_compile_definitions("sentry_tests" PUBLIC -DSENTRY_UNITTEST)
+target_link_libraries("sentry_tests" ${LINK_LIBRARIES})
+
+# to fix some issues with tests (dladdr can't find functions otherwise)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "Android")
+	target_link_libraries("sentry_tests" "-Wl,--build-id=uuid,-E")
+endif()
```

Comment:
this is causing issues on Android if building on Windows.

https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#additional-required-arguments
`uuid` is not the default form of `--build-id`, why do we need this one and not the default one?

What's about `LLD`? we could try `-fuse-ld=lld`, it's better at Windows, bfd/gold are very Linux first.

Also, not sure if a fix should come from sentry-native or sentry-android-ndk, as build was working before and now it doesn't.

Just excluding `sentry_tests` on sentry-android-ndk seems wrong to me, I'd need to do multiple times probably.

@Swatinem

Reply:
Sentry is good with any build id for as long as there is one. We used to see SHA1 or other checksums in the field.

Reply:
@Swatinem so unless you have a special reason to require `uuid` type, would it be ok to use the standard one?

Reply:
uuid is just the fastest, feel free to replace it.

Reply:
The important bit is -E

Reply:
@jan-auer I see, thanks.
https://github.com/getsentry/sentry-native/pull/134